src/tools/immutable_object_cache/ObjectCacheStore.cc

```diff
@@ -193,7 +248,7 @@ int ObjectCacheStore::lookup_object(std::string pool_nspace,
 
   switch (ret) {
     case OBJ_CACHE_NONE: {
-      pret = do_promote(pool_nspace, pool_id, snap_id, object_name);
+      pret = do_promote(pool_nspace, pool_id, snap_id, object_size, object_name);
```

Comment:
Nit: should this actually be put in the policy? What happens if the policy says "NONE" (i.e. please promote this) but it's never promoted since it was throttled?

Reply:
If the policy says  OBJ_ CACHE_ NONE, and then enter "do_promote" function. Check whether the current is limited at the beginning of the function. If the current is limited, return - 1 directly. Because there is already a promote event in execution that takes up bandwidth, it will not be "never", it is just in progress.If this logic is unreasonable, it would be better to put throttle's judgment before promote instead of In the do_promote function.

Reply:
I don't think the goal is to delay the actual IO operation -- but rather to delay the rate at which items are promoted to the cache. Put in another way, we don't want to rate-limit the librbd client that is performing the IO.

Reply:
I see what you mean. This IO should be postponed instead of cancelled, otherwise it may never be promoted.I think about what to do here. It seems that I need to create a queue or another thread to process.